client/components/forms/language-selector.jsx

```diff
@@ -1,56 +1,41 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	debug = require( 'debug' )( 'calypso:forms:language-selector' );
+import React from 'react';
+import { localize } from 'i18n-calypso';
+import { omit } from 'lodash';
+
 /**
  * Internal dependencies
  */
-var SelectOptGroups = require( 'components/forms/select-opt-groups' );
-
-function coerceToOptions( data, valueKey ) {
-	valueKey = 'undefined' === typeof valueKey ? 'value' : valueKey;
+import SelectOptGroups from 'components/forms/select-opt-groups';
 
-	return data.map( function( language ){
-		return { value: language[ valueKey ], label: language.name };
-	} );
+function coerceToOptions( data, valueKey = 'value' ) {
+	return data.map( language => ( {
+		value: language[ valueKey ],
+		label: language.name,
+	} ) );
 }
 
-var LanguageSelector = React.createClass( {
-
-	displayName: 'LanguageSelector',
-
-	componentWillMount: function() {
-		debug( 'Mounting LanguageSelector React component.' );
-	},
-
-	languageOptGroups: function() {
-		var allLanguages, popularLanguages;
+const LanguageSelector = props => {
+	const { languages, valueKey, translate } = props;
+	const allLanguages = coerceToOptions( languages, valueKey );
+	let popularLanguages = languages.filter( language => language.popular );
+	popularLanguages.sort( ( a, b ) => a.popular - b.popular );
+	popularLanguages = coerceToOptions( popularLanguages, valueKey );
 
-		allLanguages = coerceToOptions( this.props.languages, this.props.valueKey );
-
-		popularLanguages = this.props.languages.filter( function( language ) { return language.popular; } );
-		popularLanguages.sort( function( a, b ) { return a.popular - b.popular; } );
-		popularLanguages = coerceToOptions( popularLanguages, this.props.valueKey );
-
-		return [
+	const languageOptGroups = [
 		{
-			label: this.translate( 'Popular languages', { textOnly: true } ),
+			label: translate( 'Popular languages', { textOnly: true } ),
 			options: popularLanguages
 		},
 		{
-			label: this.translate( 'All languages', { textOnly: true } ),
+			label: translate( 'All languages', { textOnly: true } ),
 			options: allLanguages
-		}
-		];
-
-	},
+		},
+	];
 
-	render: function() {
-		return (
-			<SelectOptGroups optGroups={ this.languageOptGroups() } {...this.props} />
-		);
-	}
-});
+	return <SelectOptGroups optGroups={ languageOptGroups } { ...omit( props, 'languages' ) } />;
```

Comment:
Apparently, `valueKey` and `valueLink` (which is the next warning that pops up after fixing `valueKey`) are a [2-way binding feature deprecated in React 0.15](https://facebook.github.io/react/docs/two-way-binding-helpers.html).

I've never used it (actually, I've never even heard of it), but at a quick glance it seems that I can't just omit them and call it a day.
`valueLink` is a function that sets the (parent) component state and it's triggered by the `onChange` of the (children) form field.

See, for example, [what happens in this particular case](https://github.com/Automattic/wp-calypso/blob/master/client/me/account/main.jsx#L85-L100) and check out how many times `valueLink` is used throughout that file!

Now, considering that it's a deprecated feature, in my opinion we shouldn't fix the warning on the children component (I mean, that prop shouldn't be there at all), but by properly refactoring the parent one.